content/automate/ManageIQ/Transformation/Common.class/__methods__/assesstransformation.rb

```diff
@@ -53,7 +53,9 @@ def main
             raise "No destination cluster for '#{source_vm.name}'. Exiting." if destination_cluster.nil?
 
             source_ems = source_vm.ext_management_system
+            task.set_option(:source_ems_id, source_ems.id)
```

Comment:
Do you really need to store `:source/destination_ems_id`? They can be easily retrieved though one or few method calls. The `options` field is getter fatter and fatter. We should store here information that we cannot get from other places, for example the progress.

Reply:
You can create utility methods for procedures like the example you gave. Better we can move this code to backend model and its corresponding service model. 
We can merge the PR as is if it is urgently needed, but I prefer a follow-up refactoring to move some redundant information out of the options field. They should be replaced by a dedicated method.

Reply:
I agree that taking time to rethink the options hash and moving thing in the ServiceTemplateTransformationPlanTask model would be good. We'll do it for gaprindashvili-5 ?